examples/multiple-regions/linuxAmi.ts

```diff
@@ -0,0 +1,161 @@
+// Copyright 2016-2017, Pulumi Corporation.  All rights reserved.
+
+// This file defines some maps that correspond to the recommended AWS Marketplace values:
+//     http://docs.aws.amazon.com/servicecatalog/latest/adminguide/catalogs_marketplace-products.html
+// Ultimately, this lets us choose a recommended Amazon Linux AMI; see:
+//     https://aws.amazon.com/amazon-linux-ami/instance-type-matrix/
+
+import * as aws from "@pulumi/aws";
+
+/**
+ * instanceTypeArch is a map of instance type to its architecture.
+ */
+const instanceTypeArch: {
+    [instanceType: string]: string,
+} = {
+    "t1.micro"   : "PV64" ,
+    "t2.nano"    : "HVM64",
+    "t2.micro"   : "HVM64",
+    "t2.small"   : "HVM64",
+    "t2.medium"  : "HVM64",
+    "t2.large"   : "HVM64",
+    "m1.small"   : "PV64" ,
+    "m1.medium"  : "PV64" ,
+    "m1.large"   : "PV64" ,
+    "m1.xlarge"  : "PV64" ,
+    "m2.xlarge"  : "PV64" ,
+    "m2.2xlarge" : "PV64" ,
+    "m2.4xlarge" : "PV64" ,
+    "m3.medium"  : "HVM64",
+    "m3.large"   : "HVM64",
+    "m3.xlarge"  : "HVM64",
+    "m3.2xlarge" : "HVM64",
+    "m4.large"   : "HVM64",
+    "m4.xlarge"  : "HVM64",
+    "m4.2xlarge" : "HVM64",
+    "m4.4xlarge" : "HVM64",
+    "m4.10xlarge": "HVM64",
+    "c1.medium"  : "PV64" ,
+    "c1.xlarge"  : "PV64" ,
+    "c3.large"   : "HVM64",
+    "c3.xlarge"  : "HVM64",
+    "c3.2xlarge" : "HVM64",
+    "c3.4xlarge" : "HVM64",
+    "c3.8xlarge" : "HVM64",
+    "c4.large"   : "HVM64",
+    "c4.xlarge"  : "HVM64",
+    "c4.2xlarge" : "HVM64",
+    "c4.4xlarge" : "HVM64",
+    "c4.8xlarge" : "HVM64",
+    "g2.2xlarge" : "HVMG2",
+    "g2.8xlarge" : "HVMG2",
+    "r3.large"   : "HVM64",
+    "r3.xlarge"  : "HVM64",
+    "r3.2xlarge" : "HVM64",
+    "r3.4xlarge" : "HVM64",
+    "r3.8xlarge" : "HVM64",
+    "i2.xlarge"  : "HVM64",
+    "i2.2xlarge" : "HVM64",
+    "i2.4xlarge" : "HVM64",
+    "i2.8xlarge" : "HVM64",
+    "d2.xlarge"  : "HVM64",
+    "d2.2xlarge" : "HVM64",
+    "d2.4xlarge" : "HVM64",
+    "d2.8xlarge" : "HVM64",
+    "hi1.4xlarge": "HVM64",
+    "hs1.8xlarge": "HVM64",
+    "cr1.8xlarge": "HVM64",
+    "cc2.8xlarge": "HVM64",
+};
+
+/**
+ * regionArchLinuxAMI is a map from region to inner maps from architecture to the recommended Linux AMI.
+ */
+export let regionArchLinuxAMI: {
+    [region: string]: { [arch: string]: string; },
+} = {
+    "us-east-1": {
```

Comment:
Is there a reason to do hard code AMIs here rather than use `getImage` and friends?

Reply:
Laziness on my part. :)

Are you referring to https://pulumi.io/reference/pkg/nodejs/@pulumi/aws/index.html#getAmiIds?

Reply:
That's the one! That function exists from Terraform partly because it was boring to go update the docs periodically when the AMIs rolled out of availability...

Reply:
FWIW - also slightly interesting related to this:

- https://aws.amazon.com/blogs/compute/query-for-the-latest-amazon-linux-ami-ids-using-aws-systems-manager-parameter-store/
- https://www.terraform.io/docs/providers/aws/d/ssm_parameter.html

Reply:
Ended up using the parameter store API.